itest/godot/TestRunner.gd

```diff
@@ -6,101 +6,101 @@ extends Node
-		get_tree().quit(2)
-		return
-
-	var rust_runner = IntegrationTests.new()
-
-	var gdscript_suites: Array = [
-		load("res://ManualFfiTests.gd").new(),
-		load("res://gen/GenFfiTests.gd").new(),
-		load("res://InheritTests.gd").new()
-	]
-	
-	var gdscript_tests: Array = []
-	for suite in gdscript_suites:
-		for method in suite.get_method_list():
-			var method_name: String = method.name
-			if method_name.begins_with("test_"):
-				gdscript_tests.push_back(GDScriptExecutableTestCase.new(suite, method_name))
-
-	var special_case_test_suites: Array = [
-		load("res://SpecialTests.gd").new(),
-	]
-
-	for suite in special_case_test_suites:
-		for method in suite.get_method_list():
-			var method_name: String = method.name
-			if method_name.begins_with("test_"):
-				gdscript_tests.push_back(await suite.run_test(suite, method_name))
-
-	var success: bool = rust_runner.run_all_tests(
-		gdscript_tests,
-		gdscript_suites.size(),
-		allow_focus,
-		self,
-		filters
-	)
-
-	var exit_code: int = 0 if success else 1
-	get_tree().quit(exit_code)
+    # Ensure physics is initialized, for tests that require it.
+    await get_tree().physics_frame
+
+    var allow_focus := true
+    var filters: Array = []
+    var unrecognized_args: Array = []
+    for arg in OS.get_cmdline_user_args():
+        match arg:
+            "--disallow-focus":
+                allow_focus = false
+            _:
+                if not arg.begins_with("[") or not arg.ends_with("]"):
+                    unrecognized_args.push_back(arg)
+
+                var args = arg.lstrip("[").rstrip("]").split(",")
+                filters.append_array(args)
+
+    if unrecognized_args:
+        push_error("Unrecognized arguments: ", unrecognized_args)
+        get_tree().quit(2)
+        return
+
+    var rust_runner = IntegrationTests.new()
+
+    var gdscript_suites: Array = [
+        load("res://ManualFfiTests.gd").new(),
+        load("res://gen/GenFfiTests.gd").new(),
+        load("res://InheritTests.gd").new()
+    ]
+    
+    var gdscript_tests: Array = []
+    for suite in gdscript_suites:
+        for method in suite.get_method_list():
+            var method_name: String = method.name
+            if method_name.begins_with("test_"):
+                gdscript_tests.push_back(GDScriptExecutableTestCase.new(suite, method_name))
+
+    var special_case_test_suites: Array = [
+        load("res://SpecialTests.gd").new(),
+    ]
+
+    for suite in special_case_test_suites:
+        for method in suite.get_method_list():
+            var method_name: String = method.name
+            if method_name.begins_with("test_"):
+                gdscript_tests.push_back(await suite.run_test(suite, method_name))
+
+    var success: bool = rust_runner.run_all_tests(
+        gdscript_tests,
+        gdscript_suites.size(),
+        allow_focus,
+        self,
+        filters
+    )
+
+    var exit_code: int = 0 if success else 1
+    get_tree().quit(exit_code)
```

Comment:
Please revert GDScript formatting to tabs (everywhere).

Reply:
Ugh, one of these snuck in. The godot editor was auto-formatting them even if I didn't save the file in the editor. It's a real pain.